pandas/core/indexes/base.py

```diff
@@ -2005,15 +2005,37 @@ def __contains__(self, key):
             return False
 
     _index_shared_docs['contains'] = """
-        return a boolean if this key is IN the index
+        Return a boolean if this key is in the index.
 
         Parameters
         ----------
-        key : object
+        key : label
+            The key requested. Immutable-like, 1-dimensional.
```

Comment:
While true, I think users will find this a bit confusing. I think it probably makes more sense to avoid saying that is 1-dimensional, as it gives the impression that more than one key can be provided.

Personally, I think the users should look for what a label can be (immutable, 1-dimensional if a tuple...) in some other part of the documentation, and here focus more on what `contains` exactly expects (the key that will be searched in the index). But if you prefer to be specific, I'd start this by something like `The key can be of the same type as the labels in index, so it needs to be immutable...`.

Reply:
This is exactly what I tried to explain, thank you!